engine/platform/src/platform_window_glfw3.cpp

```diff
@@ -172,13 +172,74 @@ namespace dmPlatform
         return 0;
     }
 
+    static void OpenGLVersionHintToMajorMinor(PlatformOpenGLVersion version, uint32_t* major, uint32_t* minor)
+    {
+        switch(version)
+        {
+            case PLATFORM_OPENGL_VERSION_3_3:
+                *major = 3;
+                *minor = 3;
+                break;
+            case PLATFORM_OPENGL_VERSION_4_0:
+                *major = 4;
+                *minor = 0;
+                break;
+            case PLATFORM_OPENGL_VERSION_4_1:
+                *major = 4;
+                *minor = 1;
+                break;
+            case PLATFORM_OPENGL_VERSION_4_2:
+                *major = 4;
+                *minor = 2;
+                break;
+            case PLATFORM_OPENGL_VERSION_4_3:
+                *major = 4;
+                *minor = 3;
+                break;
+            case PLATFORM_OPENGL_VERSION_4_4:
+                *major = 4;
+                *minor = 4;
+                break;
+            case PLATFORM_OPENGL_VERSION_4_5:
+                *major = 4;
+                *minor = 5;
+                break;
+            case PLATFORM_OPENGL_VERSION_4_6:
+                *major = 4;
+                *minor = 6;
+                break;
+        }
+    }
+
     static PlatformResult OpenWindowOpenGL(Window* wnd, const WindowParams& params)
     {
-        glfwWindowHint(GLFW_CONTEXT_VERSION_MAJOR, 3);
-        glfwWindowHint(GLFW_CONTEXT_VERSION_MINOR, 3);
+        bool use_highest_version = params.m_OpenGLVersionHint == PLATFORM_OPENGL_VERSION_HIGHEST_AVAILABLE;
+
+        if (!use_highest_version)
+        {
+            uint32_t major, minor;
+            OpenGLVersionHintToMajorMinor(params.m_OpenGLVersionHint, &major, &minor);
+
+            glfwWindowHint(GLFW_CONTEXT_VERSION_MAJOR, major);
+            glfwWindowHint(GLFW_CONTEXT_VERSION_MINOR, minor);
+        }
+
+        if (params.m_OpenGLUseCoreProfileHint)
+        {
+            bool can_set_profile_and_fc = true;
+
+        #ifdef _WIN32
+            // Not supported on windows when requesting the default OpenGL version (which will equate to the highest available)
+            can_set_profile_and_fc = !use_highest_version;
+        #endif
+
+            if (can_set_profile_and_fc)
+            {
+                glfwWindowHint(GLFW_OPENGL_PROFILE, GLFW_OPENGL_CORE_PROFILE);
+                glfwWindowHint(GLFW_OPENGL_FORWARD_COMPAT, GL_TRUE);
+            }
```

Comment:
The "forward compatability" hint is only needed when using the core profile, so I decided to not add a specific setting for it. According to the OpenGL documentation, it's only really needed for < versions 3.3, which we don't support anyway..

Reply:
If Defold stopped supporting OpenGL 3.1, it should be announced in the news 🙌. Because the minimum was OpenGL 3.1 since Defold 1.2.181. And it seems that after the updating Windows platform to GLFW 3 the engine really stopped supporting 3.1.